src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -223,36 +239,28 @@ def test_WINNF_FT_S_FAD_2(self, config_filename):
     self._sas_admin.InjectUserId({'userId': device_c4['userId']})
 
     # Whitelist the FCCID and UserID of the devices loaded to SAS-test_harness in SAS UUT
-    for cbsdRecord in sas_test_harness_dump_records_config['cbsdRecords']:
+    for cbsdRecord in config['sasTestHarnessConfigDumpRecords']['cbsdRecords']:
       self._sas_admin.InjectFccId({'fccId': cbsdRecord['registration']['fccId']})
-      self._sas_admin.InjectUserId({'userId': cbsdRecord['registration']['userId']})
 
-    # Send a valid Registration Request for CBSD (C2) to the SAS UUT
-    # Check registration response of SAS UUT
-    # Send a valid Grant Request Message for CBSD (G2) to the SAS UUT
-    # Check grant response of G2 from SAS UUT
 
-    cbsd_ids_c2, grant_ids_c2 = self.assertRegisteredAndGranted([device_c2], [grant_g2])
-
-    # Send a valid Registration Request for CBSD (C4) to the SAS UUT
-    # Check registration response  of SAS UUT
-    # Send a valid Grant Request for CBSD (G4) to SAS UUT
-    # Check grant response of G4 from SAS UUT
-
-    cbsd_ids_c4, grant_ids_c4 = self.assertRegisteredAndGranted([device_c4], [grant_g4])
+    # Register devices C2 and C4, request grants G2 and G4 respectively with SAS UUT.
+    # Ensure the registration and grant requests are successful.
+    cbsd_ids, grant_ids = self.assertRegisteredAndGranted([device_c2, device_c4],
+                                                          [grant_g2, grant_g4],
+                                                          config['conditionals'])
```

Comment:
What will happen if no conditional registration data is specified? Please see e.g. GRA.17 for an example of how to handle this.

Reply:
Updated accordingly